fields/interface.go

```diff
@@ -0,0 +1,9 @@
+package fields
+
+type marshaler interface {
```

Comment:
could you point to some places in the standard library where this is used but not exported?

Reply:
There are similar but not exactly the same patterns in the stdlib. We looked at https://github.com/gogo/protobuf/issues/28 together. I'm still not clear on why it wasn't implemented - the issue was fairly confusing and from over 3 years ago.